### PR TITLE
fix(links): support urls without protocol

### PR DESCRIPTION
**Pull request by gadzhimari**
Update regex url pattern:

- match urls without protocol
- add support cyrillic domains

### CODE REVIEW

**Comment by meafmira** [comment]
@gadzhimari we need match only real first level domains not all. I think u can use smthg like this lib - https://www.npmjs.com/package/tlds

File: src/decorators/link.js [hunk 1]
```diff
@@ -39,16 +42,28 @@ function getBraceDepth(text: string): number {
...
     for (let matches = pattern.exec(text); matches !== null; matches = pattern.exec(text)) {
-      const name = matches[1];
+      const [, name, url, domain] = matches;
+
+      if (!domains.has(domain)) {
+        continue;
+      }
 
-      let link = matches[2];
+      let link = normalizeUrl(url);
```
Comment (meafmira) [inline src/decorators/link.js]: @gadzhimari no. You should pass normalized url in `options` field, do not replace real content

Comment (gadzhimari) [reply src/decorators/link.js]: but that variable already used in options field

File: src/decorators/link.js [hunk 2]
```diff
@@ -39,16 +42,28 @@ function getBraceDepth(text: string): number {
   return depth;
 }
 
+function urlHasHttps(url: string): boolean {
```
Comment (meafmira) [inline src/decorators/link.js]: @gadzhimari you already have this check in `pattern`-regexp. Remove

Comment (gadzhimari) [reply src/decorators/link.js]: I add it because I need a way to check out if link has http prefix to dynamically manage `options` property

**Pull request merged**